episodes/24-linear-reg-broom.Rmd

```diff
@@ -0,0 +1,323 @@
+
+### Log transform
+
+If your data is skwewed, it can be useful to transform a variable to it's log form when doing the regression.
+You can either transform the variable beforehand or do so in the equation.
+
+```{r}
+reg_logbarriers_health <- lm(health_london_rank ~ log(barriers_london_rank), data = lon_dims_imd_2019)
+
+summary(reg_logbarriers_health)
+```
+The interpretation of the log-transformed variable is a bit different.
+In this example only the predictor variable is log tranformed, therefore to interpret the slope coefficient we divide it by 100 (2917.0/100=29.170).
+
+If the dependent/response variable is solely log-transformed. Exponentiate the coefficient. This gives the multiplicative factor for every one-unit increase in the independent variable. Example: the coefficient is 0.198. exp(0.198) = 1.218962. For every one-unit increase in the independent variable, our dependent variable increases by a factor of about 1.22, or 22%. Recall that multiplying a number by 1.22 is the same as increasing the number by 22%. Likewise, multiplying a number by, say 0.84, is the same as decreasing the number by 1 – 0.84 = 0.16, or 16%.
+
+If both are transformed. nterpret the coefficient as the percent increase in the dependent variable for every 1% increase in the independent variable. Example: the coefficient is 0.198. For every 1% increase in the independent variable, our dependent variable increases by about 0.20%. For x percent increase, calculate 1.x to the power of the coefficient, subtract 1, and multiply by 100. Example: For every 20% increase in the independent variable, our dependent variable increases by about (1.20 0.198 - 1) * 100 = 3.7 percent.
+
+### Predicted values and Residuals
+
+We can expand our simple linear regression example to incorporate the Barriers to Housing and Services Domain rank.
+The Barriers to Housing and Services Domain measures the physical and financial accessibility of housing and local services. The indicators fall into two sub-domains: ‘geographical barriers’, which relate to the physical proximity of local services, and ‘wider barriers’ which includes issues relating to access to housing, such as affordability.
+
+```{r}
+reg_LivEnv_barriers_health <- lm(
+  health_london_rank ~ livingEnv_london_rank + barriers_london_rank,
+  data = lon_dims_imd_2019
+)
+
+summary(reg_LivEnv_barriers_health)
+```
+
+After running the regression model, we can access the model predicted values and the residuals compared to the real observations.
+
+```{r}
+# first we fit the predictions
+health_rank_pred <- fitted(reg_LivEnv_barriers_health)
+health_rank_pred <- as.data.frame(health_rank_pred)
+
+# now we add the residual values too
+health_rank_resid <- residuals(reg_LivEnv_barriers_health)
+health_rank_pred$resid <- health_rank_resid
+
+# We can thenview the predictions and residuals
+View(health_rank_pred)
+```
```

Comment:
This won't work in an Rmd file and shows the following error:

```
Error in check_for_XQuartz(file.path(R.home("modules"), "R_de.so")): X11 library is missing: install XQuartz from www.xquartz.org
```

Would suggest replacing this with `head(health_rank_pred)` to get a small preview and then add a separate, non-evaluated code chunk to point out that you can use this to view the data in RStudio.

````suggestion
head(health_rank_pred)
```

```{r, eval=FALSE}
# You can view the full data in RStudio with the View() function
View(health_rank_pred)
```
````